### PR TITLE
Add revamped photo gallery, quick community switcher

### PR DESCRIPTION
**Pull request by aeharding**
I'm pretty excited about this one...

### CODE REVIEW

File: src/features/gallery/GalleryPostActions.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  const postVotesById = useAppSelector((state) => state.post.postVotesById);
+  const buildGeneralBrowseLink = useBuildGeneralBrowseLink();
+  const link = buildGeneralBrowseLink(
+    `/c/${getHandle(post.community)}/comments/${post.post.id}`
+  );
+
+  const score =
+    post.counts.score -
+    (post.my_vote ?? 0) +
+    (postVotesById[post.post.id] ?? 0);
```
Comment (rsammelson) [inline src/features/gallery/GalleryPostActions.tsx:50]: Is this duplicated from elsewhere? Should there be a helper for it?

**Review by rsammelson — CHANGES_REQUESTED** [review]
I'm excited for this to land! ~~I don't understand what TitleSearch is though.~~ Edit: I figured it out.

File: src/features/comment/reply/CommentReply.tsx [hunk 2]
```diff
@@ -22,6 +22,7 @@ import { useAppSelector } from "../../../store";
 import { Centered, Spinner } from "../../auth/Login";
 import { handleSelector, jwtSelector } from "../../auth/authSlice";
 import { css } from "@emotion/react";
+import { photoswipeGalleryPreventFocusTrap as preventPhotoswipeGalleryFocusTrap } from "../../gallery/Gallery";
```
[the remark this answers is not in the document]
Comment (aeharding) [reply src/features/comment/reply/CommentReply.tsx:25]: LOL vscode

File: src/features/post/detail/PostDetail.tsx [hunk 3]
```diff
@@ -155,19 +155,13 @@ export default function PostDetail() {
...
     }
 
-    if (markdownLoneImage)
-      return (
-        <LightboxImg
-          src={markdownLoneImage.url}
-          alt={markdownLoneImage.altText}
-        />
-      );
+    if (markdownLoneImage) return <PostGallery post={post} />;
```
[the remark this answers is not in the document]
Comment (aeharding) [reply src/features/post/detail/PostDetail.tsx:164]: Damn, good catch

File: src/features/gallery/GalleryPostActions.tsx (hunk 1 above)
Comment (aeharding) [reply src/features/gallery/GalleryPostActions.tsx]: Should be much better now!

File: src/features/labels/Vote.tsx [hunk 4]
```diff
@@ -26,45 +28,37 @@ const Container = styled.div<{ vote: 1 | -1 | 0 | undefined }>`
...
-  voteFromServer,
-  score: existingScore,
-  className,
-}: VoteProps) {
+export default function Vote({ item, className }: VoteProps) {
   const [present] = useIonToast();
   const dispatch = useAppDispatch();
   const votesById = useAppSelector((state) =>
-    type === "comment"
+    "comment" in item
```
Comment (rsammelson) [inline src/features/labels/Vote.tsx:39]: What does `"comment" in item` mean exactly?

Comment (aeharding) [reply src/features/labels/Vote.tsx:39]: It infers that it's a `CommentView`, and Typescript picks this up too.

**Comment by aeharding** [comment]
@rsammelson I'm planning to get this merged in for the release shortly. Let me know if you're still reviewing.

**Comment by rsammelson** [comment]
Yeah I have one more thing.

**Comment by rsammelson** [comment]
See #292

**Comment by rsammelson** [comment]
I think I found a bug. In FF on MacOS, open an image and reload the page. The image ends up full screen.

Original vs reloaded:

original
reloaded

**Comment by aeharding** [comment]
@rsammelson Not sure why that's happening. I can remove the URL pushing stuff that's causing this to happen, but the reason this was included was to support android back button

**Comment by aeharding** [comment]
OK, I found a fix. I'll just write a uuid to the photoswipe id

**Comment by aeharding** [comment]
pushed

**Comment by rsammelson** [comment]
@aeharding I don't see a commit?

**Comment by aeharding** [comment]
Sorry, forgot to pull the merge.

**Comment by rsammelson** [comment]
Now it closes the photo on reload, but the URL is still full of image junk. Is that the intended behavior?

**Comment by aeharding** [comment]
> Now it closes the photo on reload, but the URL is still full of image junk. Is that the intended behavior?

Yeah, that's fine.

This is a fix for android back button. If you'd like I could restrict to android only. But meh

**Comment by rsammelson** [comment]
Better to close the image view than have the image fill the screen I think.

**Review by rsammelson — APPROVED** [review]

**Comment by aeharding** [comment]
Pushed a last fix to cleanup the URL a bit, so it doesn't look so sketch. (Same behavior though)

**Comment by aeharding** [comment]
Meh. I decided to just UA sniff. I hate looking at that URL on desktop, haha

**Comment by rsammelson** [comment]
Oh now it works great, thanks!

**Pull request merged**